streams/src/main/java/org/apache/kafka/streams/processor/internals/ActiveTaskCreator.java

```diff
@@ -108,19 +113,34 @@ private static String getTaskProducerClientId(final String threadClientId, final
             final String threadProducerClientId = getThreadProducerClientId(threadId);
             final Map<String, Object> producerConfigs = config.getProducerConfigs(threadProducerClientId);
 
-            if (processingMode == StreamThread.ProcessingMode.EXACTLY_ONCE_BETA) {
-                producerConfigs.put(ProducerConfig.TRANSACTIONAL_ID_CONFIG, applicationId + "-" + processId);
-                threadProducer = new StreamsProducer(clientSupplier.getProducer(producerConfigs), true, logContext);
+            if (processingMode == EXACTLY_ONCE_BETA) {
+                producerConfigs.put(ProducerConfig.TRANSACTIONAL_ID_CONFIG, transactionalId);
+                threadProducer = new StreamsProducer(clientSupplier.getProducer(producerConfigs), EXACTLY_ONCE_BETA, logContext);
             } else {
-                threadProducer = new StreamsProducer(clientSupplier.getProducer(producerConfigs), false, logContext);
+                threadProducer = new StreamsProducer(clientSupplier.getProducer(producerConfigs), AT_LEAST_ONCE, logContext);
             }
             taskProducers = Collections.emptyMap();
         }
     }
 
+    public void reInitializeThreadProducer() {
```

Comment:
A `ProducerFencedException` is thrown if a transaction expires and is aborted by the broker. Hence, it seems reasonable to handle this case gracefully and to create a new producer and rejoin the consumer group. If we don't do anything the thread won't die but would "spin" and fail over-and-over again with `ProduceFencedException` as it will try to rejoin the group but never succeed.

The only alternative would be, to terminate the thread explicitly, but this does not seem to be appropriate.